packages/contracts-bedrock/test/L1/OptimismPortal.t.sol

```diff
@@ -377,6 +397,124 @@ contract OptimismPortal_Test is CommonTest {
         vm.expectRevert();
         optimismPortal.setGasPayingToken({ _token: address(0), _decimals: 0, _name: "", _symbol: "" });
     }
+
+    function testFuzz_depositERC20Transaction_succeeds(uint256 _amount) external {
+        MockERC20 token = new MockERC20("Test", "TST", 18);
+
+        // Mint the token to the contract and approve the token for the portal
+        token.mint(address(this), _amount);
+        token.approve(address(optimismPortal), _amount);
+
+        // Mock the gas paying token to be the ERC20 token
+        vm.mockCall(address(systemConfig), abi.encodeWithSignature("gasPayingToken()"), abi.encode(address(token), 18));
+
+        vm.expectEmit(address(optimismPortal));
+        emit TransactionDeposited(
+            AddressAliasHelper.applyL1ToL2Alias(address(this)),
+            address(0),
+            uint256(0),
+            abi.encodePacked(
+                _amount, // mint
+                uint256(0), // value
+                optimismPortal.minimumGasLimit(0), // gasLimit
+                false, // isCreation,
+                "" // data
+            )
+        );
+
+        // Deposit the token into the portal
+        optimismPortal.depositERC20Transaction(address(0), _amount, 0, optimismPortal.minimumGasLimit(0), false, "");
+    }
+
+    function test_depositERC20Transaction_notCustomGas_reverts() external {
+        vm.expectRevert("OptimismPortal: only custom gas token");
+
+        optimismPortal.depositERC20Transaction(address(0), 0, 0, 0, false, "");
+    }
+
+    function test_depositERC20Transaction_isCreationNotZeroTarget_reverts() external {
+        MockERC20 token = new MockERC20("Test", "TST", 18);
+
+        // Mock the gas paying token to be the ERC20 token
+        vm.mockCall(address(systemConfig), abi.encodeWithSignature("gasPayingToken()"), abi.encode(address(token), 18));
+
+        uint64 gasLimit = optimismPortal.minimumGasLimit(0);
+
+        vm.expectRevert("OptimismPortal: must send to address(0) when creating a contract");
+
+        // Deposit the token into the portal
+        optimismPortal.depositERC20Transaction(address(1), 0, 0, gasLimit, true, "");
+    }
+
+    function test_depositERC20Transaction_gasLimitTooLow_reverts() external {
+        MockERC20 token = new MockERC20("Test", "TST", 18);
+
+        // Mock the gas paying token to be the ERC20 token
+        vm.mockCall(address(systemConfig), abi.encodeWithSignature("gasPayingToken()"), abi.encode(address(token), 18));
+
+        vm.expectRevert("OptimismPortal: gas limit too small");
+
+        // Deposit the token into the portal
+        optimismPortal.depositERC20Transaction(address(0), 0, 0, 0, false, "");
+    }
+
+    function test_depositERC20Transaction_dataTooLarge_reverts() external {
+        bytes memory data = new bytes(120_001);
+        data[120_000] = 0x01;
+
+        MockERC20 token = new MockERC20("Test", "TST", 18);
```

Comment:
All of the tests that deploy a `MockERC20` can be put into their own contract that inherits from `OptimismPortal_Test` and then deploys the token in `setup`. This will reduce code duplication